### ISSUE TITLE
nerdctl run with restart policy "unless-stopped" causes colima to hang

### ISSUE THREAD

**Issue by Nezteb** [abiosoft/colima#722]
### Description

I'm trying to run this container: https://github.com/CorentinTh/it-tools#self-host

```
❯ docker run -d --name it-tools --restart unless-stopped -p 8080:80 corentinth/it-tools:latest
docker.io/corentinth/it-tools:latest:                                             resolved       |++++++++++++++++++++++++++++++++++++++|
index-sha256:c51cffdf95ceca51e90aa51c55334a5a5c3b23b7479340e165658a32a0350f0e:    done           |++++++++++++++++++++++++++++++++++++++|
manifest-sha256:2b4397062e380449a023669f40767437167152e12cd926f1d5362a3d14918373: done           |++++++++++++++++++++++++++++++++++++++|
config-sha256:6e0cb098c4bfbd688905a521cc4ca9d975f8ce9d492e56dcdca0febc4f4e6514:   done           |++++++++++++++++++++++++++++++++++++++|
elapsed: 4.8 s                                                                    total:   0.0 B (0.0 B/s)
FATA[0004] unsupported restart policy "unless-stopped", supported policies are: []
FATA[0004] exit status 1
```

After that it kicks me back to my shell.

I was confused at this point because `nerdctl` definitely supports the `unless-stopped` restart policy: https://github.com/containerd/nerdctl/blob/main/docs/command-reference.md#container-management

If I then try to run any other `nerdctl` command, it just hangs:
```
❯ nerdctl ps

```
I have to `CTRL+C` to kill it.

If I try to stop `colima`, it also hangs:
```
❯ colima stop
INFO[0000] stopping colima

```
I have to `CTRL+C` this as well.

Then I have to run `ps aux | grep colima` and send a `kill -9` to its pid.

After manually killing `colima`, I tried `colima delete` and `colima start` to get a fresh state again. I try the original `nerdctl run` command, and the exact same thing happens. I've also tried rebooting my laptop just in case, but it still occurs. I'm not sure if it's an issue with the image I'm trying to run or what.

### Version

Running macOS 13.3.1 on an M2 Pro chip.

```
❯ colima version
colima version 0.5.4
git commit: feef4176f56a7dea487d43689317a9d7fe9de27e

runtime: containerd
arch: aarch64
client: v1.1.0
server: v1.6.8

❯ limactl --version
limactl version 0.15.1

❯ qemu-img --version
qemu-img version 8.0.0
Copyright (c) 2003-2022 Fabrice Bellard and the QEMU Project developers
```

### Operating System

- [X] macOS M1 >= 13 (Ventura)

### Output of `colima status`

After a fresh `colima start`:
```
❯ colima status
INFO[0000] colima is running using macOS Virtualization.Framework
INFO[0000] arch: aarch64
INFO[0000] runtime: containerd
INFO[0000] mountType: virtiofs
```

### Reproduction Steps

1. Use the above `nerdctl run` command (I guess with the same image?)
2. Notice the fatal error.
3. Try to run any other `nerdctl` or `colima` command.

### Expected behaviour

Ideally there should be no error (since `nerdctl` supports the `unless-stopped` restart policy), but even if there is it shouldn't cause the entire `colima` runtime to hang.

**Comment by jandubois** [comment]
This restart policy requires containerd 1.7.0 or later, which is only available in Alpine 3.18 or in the edge repos. I suspect that colima uses an older version.

**Comment by Nezteb** [comment]
@jandubois Ahh, that makes sense.

Funnily enough I had asked for a `nerdctl` bump only a few days ago, but it had already been done: https://github.com/abiosoft/alpine-lima/pull/4 😅

I stopped/deleted `colima`, ran `brew upgrade colima`, got 0.5.5, and now:
```
❯ colima version
colima version 0.5.5
git commit: 6251dc2c2c5d8197c356f0e402ad028945f0e830

runtime: containerd
arch: aarch64
client: v1.3.1
server: v1.7.0
```

So now my `nerdctl run` command works.

I also tried to pass an invalid restart policy to see if I could replicate the hanging bug: `docker run -d --name it-tools --restart foobar -p 8080:80 corentinth/it-tools:latest`

However, now things aren't hanging. I'll consider that something that was fixed in 0.5.5 as well. 😄 

Thanks for the help!
